circuitv2: flaky TestReservationFailures

https://github.com/libp2p/go-libp2p/runs/7748150365?check_suite_focus=true

```
  === RUN   TestReservationFailures
  === RUN   TestReservationFailures/unsupported_protocol
  2022/08/09 14:31:47 failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/lucas-clemente/quic-go/wiki/UDP-Receive-Buffer-Size for details.
  === RUN   TestReservationFailures/wrong_message_type
  === RUN   TestReservationFailures/unknown_status
  === RUN   TestReservationFailures/invalid_time
      reservation_test.go:101: 
          	Error Trace:	/home/runner/work/go-libp2p/go-libp2p/p2p/protocol/circuitv2/client/reservation_test.go:101
          	Error:      	"error writing reservation message: Application error 0x0: peer:QmQkiUAs4viJ4L6VYx1f4QTcc8oWcCsk8qxDxU9efYK3Fs: cannot reserve inbound connection: resource limit exceeded" does not contain "received reservation with expiration date in the past"
          	Test:       	TestReservationFailures/invalid_time
  === RUN   TestReservationFailures/invalid_voucher
  --- FAIL: TestReservationFailures (8.07s)
      --- PASS: TestReservationFailures/unsupported_protocol (2.04s)
      --- PASS: TestReservationFailures/wrong_message_type (1.53s)
      --- PASS: TestReservationFailures/unknown_status (1.77s)
      --- FAIL: TestReservationFailures/invalid_time (1.72s)
      --- PASS: TestReservationFailures/invalid_voucher (1.01s)

--- a/p2p/host/resource-manager/limit_defaults.go
+++ b/p2p/host/resource-manager/limit_defaults.go
@@ -492,7 +492,7 @@ var DefaultLimits = ScalingLimitConfig{
 	},
 
 	PeerBaseLimit: BaseLimit{
-		ConnsInbound:    4,
+		ConnsInbound:    8,
 		ConnsOutbound:   8,
 		Conns:           8,
 		StreamsInbound:  256,

--- a/p2p/host/resource-manager/limit_defaults.go
+++ b/p2p/host/resource-manager/limit_defaults.go
@@ -492,6 +492,9 @@ var DefaultLimits = ScalingLimitConfig{
 	},
 
 	PeerBaseLimit: BaseLimit{
+		// 8 for now so that it matches the number of concurrent dials we may do
+		// in swarm_dial.go. With future smart dialing work we should bring this
+		// down
 		ConnsInbound:    8,
 		ConnsOutbound:   8,
 		Conns:           8,

